internal/security/fileprovider/tokenconfig.go

```diff
@@ -80,3 +88,77 @@ func LoadTokenConfig(fileOpener fileioperformer.FileIoPerformer, path string, to
 
 	return nil
 }
+
+// GetTokenConfigFromEnv function gets a list of token service keys
+// from environment variable and populates the default configuration with
+// default token parameters and policies
+// the function returns a TokenConfFile map instance and error if any
+// if the environment variable is not present or the value of that is empty
+// then it will return empty map
+// if the value for the list is not well-formed, not comma-separated
+// then it will return an error
+func GetTokenConfigFromEnv() (TokenConfFile, error) {
+	emptyTokenConfig := make(TokenConfFile)
+
+	addTokenList := os.Getenv(addSecretstoreTokensEnvKey)
+	if strings.TrimSpace(addTokenList) == "" {
+		return emptyTokenConfig, nil
+	}
+
+	// the list of service names is comma-separated
+	tokenConfigFromEnv := make(TokenConfFile)
+	serviceNameList := strings.Split(addTokenList, ",")
+	// regex for valid service name as key
+	// the service name eventually becomes part of the URL to Vault's API call
+	// / is not ok and backslack \ is not ok comma , is not ok
+	// and the the length of name is upto 512 characters
+	serviceNameRegx := regexp.MustCompile(`^[^.*\/\\\,]*[^\W]*[\w.]{1,512}$`)
+
+	for _, name := range serviceNameList {
+		serviceName := strings.TrimSpace(name)
+		if serviceName == "" {
+			return emptyTokenConfig, errors.New("cannot have empty service name as key from environment variable")
```

Comment:
Why error out instead of just skipping empty service names?

Reply:
can do skipping. it is just the requirement issue.

Reply:
The reason I mention in in case someone puts something like this in the docker-compose in order to build a larger list out of sublists:

EXTRA_SERVICES: $VENDOR1LIST,$VENDOR2LIST,$VENDOR3LIST

Reply:
make sense